libraries/persistent_store/src/linear.rs

```diff
@@ -123,23 +123,17 @@ impl<S: Storage> Linear<S> {
     ///
     /// This function will erase the page if needed and restore the non-overwritten part. This is
     /// not power-loss resistant.
-    fn write_within(&mut self, mut index: StorageIndex, value: &[u8]) -> StorageResult<()> {
-        let page_size = self.storage.page_size();
+    fn write_within(&mut self, index: StorageIndex, value: &[u8]) -> StorageResult<()> {
```

Comment:
Is this function equivalent to:
```
self.erase_within(index, value.len());
self.write_unaligned(index, value)
```
?

Reply:
No because if the slice is already erased, it just writes.

Reply:
Oh actually we do this check there, so yes.